modules/analyse/src/main/Advice.scala

```diff
@@ -101,18 +100,21 @@ private[analyse] object MateAdvice {
     def invertMate(mate: Mate) = mate invertIf info.color.black
     def prevCp = prev.cp.map(invertCp).??(_.centipawns)
     def nextCp = info.cp.map(invertCp).??(_.centipawns)
-    MateSequence(prev.mate map invertMate, info.mate map invertMate) map { sequence =>
+    MateSequence(prev.mate map invertMate, info.mate map invertMate) flatMap { sequence =>
       import Advice.Judgement._
-      val judgment = sequence match {
-        case MateCreated if prevCp < -999 => Inaccuracy
-        case MateCreated if prevCp < -700 => Mistake
-        case MateCreated => Blunder
-        case MateLost if nextCp > 999 => Inaccuracy
-        case MateLost if nextCp > 700 => Mistake
-        case MateLost => Blunder
-        case MateDelayed => Inaccuracy
+      val judgment: Option[Advice.Judgement] = sequence match {
+        case MateCreated if prevCp < -999 => Option(Inaccuracy)
+        case MateCreated if prevCp < -700 => Option(Mistake)
+        case MateCreated => Option(Blunder)
+        case MateLost if nextCp > 999 => Option(Inaccuracy)
+        case MateLost if nextCp > 700 => Option(Mistake)
+        case MateLost => Option(Blunder)
+        case MateDelayed => None
+      }
+      judgment match {
+        case Some(j) => Option(MateAdvice(sequence, j, info, prev))
+        case None => None
```

Comment:
or:
```judgment map { j => MateAdvice(sequence, j, info, prev) }```

Reply:
Ah, that's an improvement, I'll update my patch.